### DISCUSSION TITLE
Is it possible to obtain a key from connected database?

### DISCUSSION THREAD

**Discussion by little-brother** [utelle/SQLite3MultipleCiphers#35] [General]
I need a key to open a new connection or to attach a database in an another connection (that can be later than an original database was opened).

`pragma key` returns `SQL logic error`.
`pragma <db alias>.key` returns no row.

Or do I need to store the keys myself?

**Comment by utelle** [comment]
With `pragma key='passphrase';` you can only **set** the _passphrase_ to be used for encrypting and decrypting the database. That is, it is expected behaviour that `PRAGMA key;` (without a passphrase) fails.

The _passphrase_ is **never** stored in the database itself. It is not even saved in memory, unless you use URI parameters with the database file name or you save it in your own application.

Therefore, it is the user's responsibility to provide the correct _passphrase_ on opening an existing database file. 

There is one exception,: the SQL command `ATTACH`. If the main database is encrypted then attaching another database to the main database connection via `ATTACH` without the `KEY` parameter assumes implicitly that the passphrase of the main database is valid for the attached database, too. If the _passphrase_ of the attached database is not identical to that of the main database, then using the `KEY` parameter of the `ATTACH` command is mandatory. If the attached database is unencrypted, then specify an empty passphrase for the `KEY` parameter.

**Reply by little-brother** [reply]
Thanks a lot for the quick and detailed answer! :)
You saved my time by adding info about attacment database.
